src/test/java/seedu/dietbook/calculator/CalculatorTest.java

```diff
@@ -2,63 +2,65 @@
 
 class CalculatorTest {
 
-
-
     @Test
     void calculateCalorie_foodListOfThreeItems_sumOfCalorie() {
-        ArrayList<Food> foodList = new ArrayList<>();
-        foodList.add(new Food("chicken rice", 666, 55, 30, 0));
-        foodList.add(new Food("pancake", 150, 16, 0, 0));
-        foodList.add(new Food("bao", 290, 0, 16, 0));
-        Calculator calculator = new Calculator(foodList);
+        FoodList foodList = new FoodList();
+        foodList.addFood(1, new Food("chicken rice", 666, 55, 30, 0));
+        foodList.addFood(1, new Food("pancake", 150, 16, 0, 0));
+        foodList.addFood(1, new Food("bao", 290, 0, 16, 0));
+        CalculatorData data = new CalculatorData();
+        data.inputData(foodList);
+        Calculator calculator = new Calculator(data);
         assertEquals(666 + 150 + 290, calculator.calculateCalorie());
     }
 
     @Test
     void calculateCarb_foodListOfThreeItems_sumOfCarb() {
-        ArrayList<Food> foodList = new ArrayList<>();
-        foodList.add(new Food("chicken rice", 666, 55, 30, 0));
-        foodList.add(new Food("pancake", 150, 16, 0, 0));
-        foodList.add(new Food("bao", 290, 0, 16, 0));
-        Calculator calculator = new Calculator(foodList);
+        FoodList foodList = new FoodList();
+        foodList.addFood(1, new Food("chicken rice", 666, 55, 30, 0));
+        foodList.addFood(1, new Food("pancake", 150, 16, 0, 0));
+        foodList.addFood(1, new Food("bao", 290, 0, 16, 0));
+        CalculatorData data = new CalculatorData(foodList);
+        Calculator calculator = new Calculator(data);
         assertEquals(55 + 16, calculator.calculateCarb());
     }
 
     @Test
     void calculateProtein_foodListOfThreeItems_sumOfProtein() {
-        ArrayList<Food> foodList = new ArrayList<>();
-        foodList.add(new Food("chicken rice", 666, 55, 30, 0));
-        foodList.add(new Food("pancake", 150, 16, 0, 0));
-        foodList.add(new Food("bao", 290, 0, 16, 0));
-        Calculator calculator = new Calculator(foodList);
+        FoodList foodList = new FoodList();
+        foodList.addFood(1, new Food("chicken rice", 666, 55, 30, 0));
+        foodList.addFood(1, new Food("pancake", 150, 16, 0, 0));
+        foodList.addFood(1, new Food("bao", 290, 0, 16, 0));
+        CalculatorData data = new CalculatorData(foodList);
+        Calculator calculator = new Calculator(data);
         assertEquals(30 + 16, calculator.calculateProtein());
     }
 
     @Test
     void calculateFat_foodListOfThreeItems_sumOfFat() {
-        ArrayList<Food> foodList = new ArrayList<>();
-        foodList.add(new Food("chicken rice", 666, 55, 30, 0));
-        foodList.add(new Food("pancake", 150, 16, 0, 0));
-        foodList.add(new Food("bao", 290, 0, 16, 0));
-        Calculator calculator = new Calculator(foodList);
+        FoodList foodList = new FoodList();
+        foodList.addFood(1, new Food("chicken rice", 666, 55, 30, 0));
```

Comment:
Consider adding tests with portion size >1. This would have caught the potential issue raised in earlier comments about portion sizing.

Reply:
OK, I changed the corresponding methods.